src/main/java/net/cytonic/cytosis/data/Database.java

```diff
@@ -0,0 +1,108 @@
+package net.cytonic.cytosis.data;
+
+import net.cytonic.cytosis.config.CytosisSettings;
+import net.cytonic.cytosis.logging.Logger;
+import net.minestom.server.MinecraftServer;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Database {
+
+    private final ExecutorService worker;
+    private final String host;
+    private final int port;
+    private final String database;
+    private final String username;
+    private final String password;
+    private final boolean ssl;
+    private Connection connection;
+
+    public Database() {
+        this.worker = Executors.newSingleThreadExecutor(Thread.ofVirtual().name("CytosisDatabaseWorker").uncaughtExceptionHandler((t, e) -> Logger.error(STR."An uncaught exception occoured on the thread: \{t.getName()}", e)).factory());
+        this.host = CytosisSettings.DATABASE_HOST;
+        this.port = CytosisSettings.DATABASE_PORT;
+        this.database = CytosisSettings.DATABASE_NAME;
+        this.username = CytosisSettings.DATABASE_USER;
+        this.password = CytosisSettings.DATABASE_PASSWORD;
+        this.ssl = CytosisSettings.DATABASE_USE_SSL;
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            Logger.error("Failed to load database driver");
+            Logger.error(e.toString());
+        }
+    }
+
+    public boolean isConnected() {
+        return (connection != null);
+    }
+
+    public void connect() {
+        worker.submit(() -> {
+            if (!isConnected()) {
+                try {
+                    connection = DriverManager.getConnection(STR."jdbc:mysql://\{host}:\{port}/\{database}?useSSL=\{ssl}&autoReconnect=true", username, password);
+                } catch (SQLException e) {
+                    Logger.error("Invalid Database Credentials!");
```

Comment:
Include the throwable in this since it might not actually be invalid credentials